packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-feedback-modal.tsx

```diff
@@ -0,0 +1,243 @@
+ * @part buttons - The wrapper around the cancel and submit buttons.
+ * @part cancel-button - The cancel button.
+ * @part submit-button - The submit button.
+ */
+@Component({
+  tag: 'atomic-smart-snippet-feedback-modal',
+  styleUrl: 'atomic-smart-snippet-feedback-modal.pcss',
+  shadow: true,
+})
+export class AtomicSmartSnippetFeedbackModal implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  @Element() public host!: HTMLElement;
+  public smartSnippet!: SmartSnippet;
+
+  @State() public error!: Error;
+
+  @Prop({mutable: true}) source?: HTMLElement;
+  @Prop({reflect: true, mutable: true}) isOpen = false;
+
+  @State() currentAnswer?: SmartSnippetFeedback | 'other' | undefined;
+  private detailsInputRef?: HTMLTextAreaElement;
+  private readonly formId = randomID(
+    'atomic-smart-snippet-feedback-modal-form-'
+  );
+
+  @Watch('isOpen')
+  watchToggleOpen(isOpen: boolean) {
+    if (isOpen) {
+      this.smartSnippet.openFeedbackModal();
+      this.currentAnswer = undefined;
+    }
+  }
+
+  initialize() {
+    this.smartSnippet = buildSmartSnippet(this.bindings.engine);
+  }
+
+  private sendFeedback() {
+    if (this.currentAnswer === 'other') {
+      this.smartSnippet.sendDetailedFeedback(this.detailsInputRef!.value); // TODO: Add textarea
+    } else {
+      this.smartSnippet.sendFeedback(this.currentAnswer!);
+    }
+    this.isOpen = false;
+  }
+
+  private close() {
+    this.isOpen = false;
+    this.smartSnippet.closeFeedbackModal();
+  }
+
+  private renderHeader() {
+    return (
+      <h1 slot="header">
+        {this.bindings.i18n.t('smart-snippet-feedback-explain-why')}
+      </h1>
+    );
+  }
+
+  private renderOptions() {
+    const options: {
+      id: string;
+      label: string;
+      isChecked: boolean;
+      onCheck: () => void;
+    }[] = [
+      {
+        id: 'does-not-answer',
+        label: this.bindings.i18n.t(
+          'smart-snippet-feedback-reason-does-not-answer'
+        ),
+        isChecked: this.currentAnswer === 'does_not_answer',
+        onCheck: () => (this.currentAnswer = 'does_not_answer'),
+      },
+      {
+        id: 'partially-answers',
+        label: this.bindings.i18n.t(
+          'smart-snippet-feedback-reason-partially-answers'
+        ),
+        isChecked: this.currentAnswer === 'partially_answers',
+        onCheck: () => (this.currentAnswer = 'partially_answers'),
+      },
+      {
+        id: 'was-not-a-question',
+        label: this.bindings.i18n.t(
+          'smart-snippet-feedback-reason-was-not-a-question'
+        ),
+        isChecked: this.currentAnswer === 'was_not_a_question',
+        onCheck: () => (this.currentAnswer = 'was_not_a_question'),
+      },
+      {
+        id: 'other',
+        label: this.bindings.i18n.t('smart-snippet-feedback-reason-other'),
+        isChecked: this.currentAnswer === 'other',
+        onCheck: () => (this.currentAnswer = 'other'),
+      },
```

Comment:
do you think a function here would be nice? like:
```js
    const availableOpts = [{id: 'does-not-answer', i18n: 'smart-snippet-feedback-reason-does-not-answer'}, ...]
    const options = availableOpts.map(({id, i18n}) => ({
      id,
      label: this.bindings.i18n.t(i18n),
      isChecked: this.currentAnswer === id,
      onCheck: () => (this.currentAnswer = id)
    }))
```

Reply:
sure! I'll do it below